pvl/math/prime_sieve.hpp

```diff
@@ -0,0 +1,28 @@
+#ifndef PVL_MATH_PRIME_SIEVE_HPP_
+#define PVL_MATH_PRIME_SIEVE_HPP_ 1
```

Comment:
- [ ] move this file to `pvl/math/numtheory/` then fix the guard definition (maybe to `PVL_MATH_NUMTHEORY_SIEVE_HPP_`?)